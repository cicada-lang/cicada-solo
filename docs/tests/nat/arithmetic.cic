function add(x: Nat, y: Nat): Nat {
  return nat_ind(y, (_) => Nat, x, (prev, almost) => add1(almost))
}

function mul(x: Nat, y: Nat): Nat {
  return nat_ind(y, (_) => Nat, 0, (prev, almost) => add(almost, x))
}

function pow(x: Nat, y: Nat): Nat {
  return nat_ind(y, (_) => Nat, 1, (prev, almost) => mul(almost, x))
}

let eight = {
  let four = add(2, 2)
  return add(four, four)
}

the(Equal(Nat, eight, pow(2, 3)), refl)
