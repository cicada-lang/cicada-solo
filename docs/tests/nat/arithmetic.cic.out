refl: Equal(Nat, 8, 8)